src/Iterator/DependencyModuleIterator.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace RebelCode\Modular\Iterator;
+
+use Dhii\Modular\Module\ModuleInterface;
+use RebelCode\Modular\Module\ModuleInterface as RcModuleInterface;
+use Iterator;
+
+/**
+ *  A module iterator that handles module dependencies.
+ *
+ * @since [*next-version*]
+ */
+class DependencyModuleIterator extends AbstractDependencyModuleIterator implements Iterator
+{
+    /**
+     * Constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param array $modules An array of modules.
+     */
+    public function __construct($modules = array())
+    {
+        $this->items = $modules;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @since [*next-version*]
+     */
+    protected function _getModuleDependencies(ModuleInterface $module)
+    {
+        if (! $module instanceof RcModuleInterface) {
+            return array();
+        }
+
+        $_this     = $this;
```

Comment:
No longer need to do this since [PHP 5.4](http://docs.php.net/manual/en/migration54.new-features.php).